experimental/status/status_source.go

```diff
@@ -174,6 +176,10 @@ func isDNSNotFoundError(err error) bool {
 	return false
 }
 
+func isSocksProxyError(err error) bool {
+	return err != nil && strings.Contains(err.Error(), "socks connect")
```

Comment:
can you check if `isConnectionResetOrRefusedError` can be used for this? I see there is an error `syscall.ENETUNREACH` which could be this case but not sure if the error you are seeing can be matched with that.